src/commonMain/kotlin/exchange.dydx.abacus/state/model/TradingStateMachine.kt

```diff
@@ -81,6 +82,8 @@ open class TradingStateMachine(
     private val maxSubaccountNumber: Int,
     private val useParentSubaccount: Boolean,
 ) {
+    private val internalState: InternalState = InternalState()
```

Comment:
should we just take this opportunity to move away from a single monolithic state object?

Reply:
It doesn't hurt to have an object that aggregates different sub states.  It'd make the migration a bit easier since it's a one-to-one mapping to the state dictionary.

Reply:
yeah i guess a couple thoughts: 

if we want to break down the state machine into smaller pieces, it'd be easier if the different components were exposed directly rather than through a aggregate object. but maybe we can cross that bridge when we get there.

also, it seems a bit strange that `internalState` itself is a `val`, but holds mutable sub properties. IMO, it'd be more clear if either:
a) `internalState` itself was a `var` and its sub fields were immutable
b) we just moved the sub fields to be come top-level mutable fields (basically get rid of `InternalState`)

Reply:
Updated...

I'd rather delay immutable implementation until we switch all internal states to static types.  copy() has a performance overhead, and we had run into performance issue before.  We can only benchmark it after all states have been converted.   Also, the current mutable implementation mirrors 1-to-1 to the state dictionary, so it's a relatively safer refactor.  Even so, there is already substantial complexity of finding all the usage of the fields within the code-base.  Switching to immutable at this stage would make the refactor riskier.